osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,672 @@
+  r["datetime"] = dumpTimestamp.str();
+  return Status();
+}
+
+Status logOSVersion(IDebugControl5* control, Row& r) {
+  unsigned long platformID;
+  unsigned long majorVersion;
+  unsigned long minorVersion;
+  unsigned long buildNumber;
+
+  if (control->GetSystemVersionValues(
+          &platformID, &majorVersion, &minorVersion, nullptr, &buildNumber) ==
+      S_OK) {
+    r["major_version"] = INTEGER(majorVersion);
+    r["minor_version"] = INTEGER(minorVersion);
+    r["build_number"] = INTEGER(buildNumber);
+    return Status();
+  }
+  return Status(1);
+}
+
+Status logDumpType(IDebugControl5* control, Row& r) {
+  unsigned long flags = 0;
+  if (control->GetDumpFormatFlags(&flags) != S_OK) {
+    return Status(1);
+  }
+
+  std::ostringstream activeFlags;
+  bool firstString = true;
+  // Loop through MINIDUMP_TYPE flags and log the ones that are set
+  for (auto const& flag : kMinidumpTypeFlags) {
+    if (flags & flag.first) {
+      if (!firstString) {
+        activeFlags << ",";
+      }
+      firstString = false;
+      activeFlags << flag.second;
+    }
+  }
+  r["type"] = activeFlags.str();
+  return Status();
+}
+
+// Note: appears to only detect unmanaged stack frames.
+// See http://blog.steveniemitz.com/building-a-mixed-mode-stack-walker-part-2/
+Status logStackTrace(IDebugControl5* control, IDebugSymbols3* symbols, Row& r) {
+  CONTEXT context = {0};
+  unsigned long type = 0;
+  unsigned long procID = 0;
+  unsigned long threadID = 0;
+  unsigned long contextSize = 0;
+  unsigned long numFrames = 0;
+  DEBUG_STACK_FRAME_EX stackFrames[kNumStackFramesToLog] = {0};
+
+  // Get stack frames, either with or without event context
+  if (control->GetStoredEventInformation(&type,
+                                         &procID,
+                                         &threadID,
+                                         &context,
+                                         sizeof(context),
+                                         &contextSize,
+                                         nullptr,
+                                         0,
+                                         0) == S_OK) {
+    symbols->SetScopeFromStoredEvent();
+    if (control->GetContextStackTraceEx(&context,
+                                        contextSize,
+                                        stackFrames,
+                                        kNumStackFramesToLog,
+                                        nullptr,
+                                        0,
+                                        0,
+                                        &numFrames) != S_OK) {
+      return Status(1);
+    }
+  } else {
+    if (control->GetStackTraceEx(
+            0, 0, 0, stackFrames, kNumStackFramesToLog, &numFrames) != S_OK) {
+      return Status(1);
+    }
+  }
+
+  // Then, log the stack frames
+  std::ostringstream stackTrace;
+  auto firstFrame = true;
+  for (unsigned long frame = 0; frame < numFrames; frame++) {
```

Comment:
Might be able to do something similar here, where you `join` a vector of `trace` values? Also prefer the use of `std::vector<char>` over `char[]` if possible. Maybe something like the following?
```
  std::vector<std::string> stackTraces;
  std::vector<char> name(512, 0x0);
  std::ostringstream trace;
  unsigned long long offset = 0;
  for (unsigned long frame = 0; frame < numFrames; frame++) {
    if (symbols->GetNameByOffset(stackFrames[frame].InstructionOffset,
      name.data(),
      static_cast<unsigned long>(name.size() - 1),
      nullptr,
      &offset) == S_OK) {
      trace << name.data() << "+0x" << std::hex << offset;
    }

    trace << "(0x" << std::hex << stackFrames[frame].InstructionOffset;
    trace << ")";
    stackTraces.push_back(trace.str());
    name.clear();
    trace.clear();
  }

  r["stack_trace"] = osquery::join(stackTraces, ",");
```